ibis/expr/analysis.py

```diff
@@ -343,26 +345,30 @@ def _lift_TableColumn(self, expr, block=None):
 
         if isinstance(root, ops.Selection):
             can_lift = False
+            all_simple_columns = all(
+                isinstance(sel.op(), ops.TableColumn)
+                and sel.op().name == sel.get_name()
+                for sel in root.selections
+                if isinstance(sel, ir.ValueExpr)
+                if sel.has_name()
+            )
```

Comment:
This is the fix for the issue. This predicate prevents lifting columns that are projected out of `Selections` with anything but simple column accesses.